numpy/linalg/_linalg.py

```diff
@@ -3174,6 +3174,142 @@ def cross(x1, x2, /, *, axis=-1):
+        the scalar cross-product must be the same size as the respective axis
+        in ``x1``.
+    axis : int, optional
+        The axis (dimension) of ``x1`` and ``x2`` containing the vectors for
+        which to compute the scalar cross-product. Default: ``-1``.
+
+    Returns
+    -------
+    out : ndarray
+        An array containing the scalar cross products.
+
+    Raises
+    ------
+    ValueError
+        When the dimension of the vector(s) in `x1` and/or `x2` does not
+        equal 2.
+
+    See Also
+    --------
+    cross : Vector cross product of 3 dimensional vectors
+    inner : Inner product
+    outer : Outer product
+    linalg.det: A similar floating point option
+
+    Examples
+    --------
+    Scalar cross product of 2-dimensional vectors:
+
+    >>> x = [1, 2]
+    >>> y = [4, 5]
+    >>> np.linalg.cross2d(x, y)
+    array(-3)
+
+    Multiple scalar cross products. Note the direction of the cross
+    product is defined by the *right-hand rule*, and
+    the cross product of a vector with itself is zero.
+
+    >>> x = np.array([[1, 2], [3, 4], [5, 6]])
+    >>> y = np.array([[3, 4], [1, 2], [5, 6]])
+    >>> np.linalg.cross2d(x, y)
+    array([-2,  2,  0])
+
+    The choice of axis matters:
+
+    >>> u = [[1, 2], [3, 4]]
+    >>> v = [[5, 6], [7, 8]]
+    >>> np.linalg.cross2d(u, v)
+    array([-4, -4])
+    >>> np.linalg.cross2d(u, v, axis=0)
+    array([-8, -8])
+
+    Compare `~linalg.cross2d` with `~linalg.det`:
+
+    >>> u = [[1, 2], [3, 4], [-1, 5]]
+    >>> v = [[5, 6], [7, 8], [-2, 9]]
+    >>> np.linalg.cross2d(u, v)
+    array([-4, -4,  1])
+    >>> stacked_array = np.stack((u, v), axis=1)
+    >>> np.linalg.det(stacked_array)
+    array([-4., -4.,  1.])
+
+    """
+    x1 = asarray(x1)
+    x2 = asarray(x2)
+
+    axis1, axis2 = (axis,) * 2
+ 
+    if (x1.ndim < 1) or (x2.ndim < 1):
+        raise ValueError("At least one array has zero dimension")
+
+    # Check axis_x1 and axis_x2 are within bounds
+    axis1 = normalize_axis_index(axis1, x1.ndim, msg_prefix='x1')
+    axis2 = normalize_axis_index(axis2, x2.ndim, msg_prefix='x2')
+
+    # Move working axis to the end of the shape
+    x1 = moveaxis(x1, axis1, -1)
+    x2 = moveaxis(x2, axis2, -1)
+
+    if x1.shape[-1] != 2 or x2.shape[-1] != 2:
+        raise ValueError(
+            "Both input arrays must be (arrays of) 2-dimensional vectors, "
+            f"but they are {x1.shape[-1]} and {x2.shape[-1]} "
+            "dimensional instead."
+        )
+
+    # Create the output array
+    shape = broadcast(x1[..., 0], x2[..., 0]).shape
+    dtype = promote_types(x1.dtype, x2.dtype)
+    cp = empty(shape, dtype)
+
+    # recast arrays as dtype
+    x1 = x1.astype(dtype)
+    x2 = x2.astype(dtype)
+
+    # create local aliases for readability
+    x1_0 = x1[..., 0]
+    x1_1 = x1[..., 1]
+    x2_0 = x2[..., 0]
+    x2_1 = x2[..., 1]
+
+    # x1_0 * x2_1 - x1_1 * x2_0
+    multiply(x1_0, x2_1, out=cp)
```

Comment:
Is there some particular reason you're not letting this line create/allocate `cp`?  At first glance it would appear to have the same result.

Reply:
I wanted to follow the code in `cross` as closely as possible, so that errors in one could be traced to errors in the others. For a really high dimensional array (with appropriate dim 2 on the correct axis), I wasn't sure how much to simplify the existing code from `cross`, so I kept it all. Here's the relevant bit from the creation of `cp` in `cross`. 

https://github.com/numpy/numpy/blob/ee73d0d9e75cae2ad09f328ce34ffd2fe9498dab/numpy/_core/numeric.py#L1655-L1687

I just trimmed off the parts that had no relevance, and swapped things to `x1` and `x2` (different naming conventions in `linalg`). I can see that defining `cp` this way in `cross` is needed because it checks `cp.ndim` first, a check not needed in `cross2d`. 
 
If I'm understanding correctly, you are suggesting the following:
- Remove L3292-L3295 entirely (as well as the import statements for `broadcast` and `promote_type`)
- Change L3308 to be `cp = x1_1 * x2_0`.  

Sounds completely reasonable.

Reply:
Looks like `promote_types` may need to stay, unless it's not needed to recast `x1` and `x2` as `dtype`. That still removes `broadcast` and creating an empty `cp`.

Reply:
Looks like none of this is needed at all, with the one-liner update from @seberg.